utils/mocks/bank.go

```diff
@@ -56,6 +58,10 @@ func NoOpSendRestrictionFn(_ sdk.Context, _, toAddr sdk.AccAddress, _ sdk.Coins)
 	return toAddr, nil
 }
 
+func FailingSendRestrictionFn(_ sdk.Context, _, addr sdk.AccAddress, _ sdk.Coins) (sdk.AccAddress, error) {
+	return nil, fmt.Errorf("%s is blocked from sending/receiving", addr.String())
+}
```

Comment:
Small nit, let's keep the same naming as the `SendRestrictionFn` interface!

```suggestion
func FailingSendRestrictionFn(_ sdk.Context, _, toAddr sdk.AccAddress, _ sdk.Coins) (sdk.AccAddress, error) {
	return nil, fmt.Errorf("%s is blocked from sending/receiving", toAddr.String())
}
```